add rsc syntax

zainin opened the pull request:
Add syntax file for RouterOS script.

Let me know if there's any issues to fix before merge.

zainin committed:
add rsc syntax

chrisbra commented on runtime/ftplugin/rsc.vim:
```diff
@@ -0,0 +1,6 @@
+" Maintainer:   zainin
+" Last Change:  2017-07-26
+
+au BufRead,BufNewFile *.rsc set filetype=rsc
```
This should go into filetype.vim, not into a filetype plugin.

zainin replied:
My bad, fixed.

chrisbra commented on runtime/syntax/rsc.vim:
```diff
@@ -0,0 +1,82 @@
+
+syn case ignore
+
+set iskeyword=A-Z,a-z
```
this needs to be `:syn iskeyword`

zainin replied:
Turns out it wasn't really necessary, so I've removed this line.

dkearns replied:
There's a few hyphenated keywords like `as-value` that are being missed because `'iskeyword'` doesn't include `-`.  If you don't want to use `syn-iskeyword` you could make them a `syn-match`.

zainin replied:
Sorry, I didn't notice `-` isn't included by default. It should be better now.

zainin committed:
rsc syntax: fix syntax and filetype detection

zainin commented on src/testdir/test_filetype.vim:
```diff
@@ -422,6 +422,7 @@ let s:filename_checks = {
     \ 'robots': ['robots.txt'],
     \ 'rpcgen': ['file.x'],
     \ 'rpl': ['file.rpl'],
+    \ 'rsc': ['file.rsc'],
```
@dkearns , it's already here, or am I missing something?

dkearns replied:
Nope, it's there alright.  I assumed it would be in the second commit and didn't even check the first.

Sorry for wasting your time.

zainin committed:
rsc syntax: fix keywords, improve file detection

zainin committed:
update CODEOWNERS for rsc.vim

dkearns committed:
Add escape sequence highlighting

dkearns committed:
Add and/or/in operator highlighting

Simplify other operator patterns and exclude surrounding whitespace from
operators.

dkearns committed:
Add new rsc filetype plugin

dkearns committed:
Exclude leading whitespace from comment highlighting

This arguably looks a bit nicer if a background colour is used for the
comment highlight group.

dkearns commented on runtime/syntax/rsc.vim:
```diff
@@ -11,6 +11,8 @@ endif
 
 syn case ignore
 
+syn iskeyword @,-
```
You've missed digits used in, for example, `toip6`.  So at a minimum I think you'll need:
```vim
syn iskeyword @,48-57,-
```

zainin replied:
You're right, fixed.

zainin committed:
rsc syntax: match keywords with digits

dkearns committed:
Remove rscSpecial delimiter highlighting from strings

These characters have no special significance within a string.

Also removes option L string prefix.  I'm assuming this is a copy/paste
error from the C syntax file as there's nothing to suggest it has or has
ever had any meaning in RouterOS Script.

dkearns committed:
Highlight the line continuation character

dkearns committed:
Rename rsc filetype to routeros

zainin committed:
Merge pull request #1 from dkearns/improve-rsc-syntax-highlighting

Improve rsc syntax highlighting

zainin committed:
add routerous maintainer contact
